### PR TITLE
TMI2-396: Add submissions to SQS for processing in Spotlight

### PR DESCRIPTION
**Pull request by GavCookCO**
Description
Adds functionality to send messages to Amazon SQS so that due diligence checks can be automatically performed. 

Story
https://technologyprogramme.atlassian.net/jira/software/projects/TMI2/boards/331?selectedIssue=TMI2-396

This PR depends on the PR below also being approved and merged: 
https://github.com/cabinetoffice/gap-find-admin-backend/pull/70

### CODE REVIEW

**Comment by GavCookCO** [comment]
The tests on this will fail while it is in draft as there is a hard coded assertion failure so that I don't forget to add important functionality when I come back to it

File: src/main/java/gov/cabinetoffice/gap/applybackend/service/GrantMandatoryQuestionService.java [hunk 1]
```diff
@@ -38,7 +38,11 @@ public GrantMandatoryQuestions getGrantMandatoryQuestionById(UUID id, String app
         return grantMandatoryQuestion.get();
     }
 
-    public GrantMandatoryQuestions getGrantMandatoryQuestionBySubmissionId(UUID submissionId, String applicantSub) {
+    /*
+        TODO I think we should decouple access control and CRUD functionality.
+        These methods should not require an applicant ID to be passed in to confirm ownership.  
+     */
+    public GrantMandatoryQuestions getGrantMandatoryQuestionBySubmissionIdAndApplicantSub(UUID submissionId, String applicantSub) {
```
Comment (iaincooper-tco) [inline src/main/java/gov/cabinetoffice/gap/applybackend/service/GrantMandatoryQuestionService.java:45]: Agreed, we do this elsewhere don't we by getting the ID from the Principal I think?

File: src/main/java/gov/cabinetoffice/gap/applybackend/config/SpotlightQueueConfigProperties.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Configuration
+@ConfigurationProperties(prefix = "spotlight-queue")
+public class SpotlightQueueConfigProperties {
+
+    @NotNull
```
Comment (GavCookCO) [inline src/main/java/gov/cabinetoffice/gap/applybackend/config/SpotlightQueueConfigProperties.java:18]: Not sure this even works. Didn't throw an error when I spelled the property name wrong...

File: src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    private final SpotlightQueueConfigProperties spotlightQueueProperties;
+    private final Clock clock;
+
+    public void createSpotlightCheck(GrantMandatoryQuestions mandatoryQuestions, GrantScheme scheme) {
+
+        // save a spotlight submission object to the database
+        final SpotlightSubmission spotlightSubmission = SpotlightSubmission.builder()
+                .mandatoryQuestions(mandatoryQuestions)
+                .grantScheme(scheme)
+                .version(scheme.getVersion()) //TODO is this how we want to set the version?
```
Comment (iaincooper-tco) [inline src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java:36]: I think we should decouple the spotlight submission version from the scheme version. There is only one version right now, so I think this should be a static value of `1`

File: src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        final UUID messageId = UUID.randomUUID();
+
+        final MessageAttributeValue spotlightSubmissionIdAttribute =  new MessageAttributeValue()
+                .withDataType("String")
+                .withStringValue(savedSpotlightSubmission.getId().toString());
+
+        final SendMessageRequest messageRequest = new SendMessageRequest()
+                .withQueueUrl(spotlightQueueProperties.getQueueUrl())
+                .withMessageGroupId(messageId.toString())
+                .withMessageBody(messageId.toString()) //TODO should we send the Spotlight Submission ID in this field instead?
```
Comment (iaincooper-tco) [inline src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java:52]: Yeah I think so, the message body needs to contain the information the spotlight worker can use to access the database records to perform the batching job, so I think the spotlight submission ID is sufficient here.

File: src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                .withMessageDeduplicationId(messageId.toString())
+                .addMessageAttributesEntry(SPOTLIGHT_SUBMISSION_ID, spotlightSubmissionIdAttribute);
+
+        amazonSqs.sendMessage(messageRequest);
+    }
+}
```
Comment (iaincooper-tco) [inline src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java:58]: Also need to set the status of the Spotlight Submission to QUEUED

File: src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java (hunk 3 above)
Comment (GavCookCO) [reply src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java:36]: Awesome, will make that change now

File: src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java (hunk 5 above)
Comment (GavCookCO) [reply src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java:58]: It goes to QUEUED by default when the object is created

File: src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java (hunk 4 above)
Comment (GavCookCO) [reply src/main/java/gov/cabinetoffice/gap/applybackend/service/SpotlightService.java]: done

**Review by iaincooper-tco — APPROVED** [review]

**Pull request merged**